### ISSUE TITLE
Cannot generate icons - error

### ISSUE THREAD

**Issue by ghost** [fomantic/create-fomantic-icons#308]
```
Fomantic-UI Icon CLI

Which icons set would you like to generate?
  > FontAwesome 6+ Free

Which version of FontAwesome 6+ Free should we download? 
  > 6.4.2

Where would you like the dist saved? [/var/www/lib/create-fomantic-icons/fui-icons]

✔ asset already exists
⠋ parsing iconsundefined:1
.4-10.1 89.4-64.4 0-61-77.1-89.6-116.9-44.6-23.5 26.4-17.6 42.1-17.6 157.6 50.7 31 118.3 22 160.4-20.1 24.8-24.8 38.5-58 38.5-93 0-35.2-13.8-68.2-38.8-93.3-24.8-24.8-57.8-38.5-93.3-38.5s-68.8 13.8-93.5 38.5c-.3.3-16 16.5-21.2 23.9l-.5.6c-3.3 4.7-6.3 9.1-20.1 6.1-6.9-1.7-14.3-5.8-14.3-11.8V20c0-5 3.9-10.5 10.5-10.5h241.3c8.3 0 8.3 11.6 8.3 15.1 0 3.9 0 15.1-8.3 15.1H130.3v132.9h.3c104.2-109.8 282.8-36 282.8 108.9 0 178.1-244.8 220.3-310.1 62.8zm63.3-260.8c-.5 4.2 4.6 24.5 14.6 20.6C306 56.6 384 144.5 390.6 144.5c4.8 0 22.8-15.3 14.3-22.8-93.2-89-234.5-57-238.3-38.2zM393 414.7C283 524.6 94 475.5 61 310.5c0-12.2-30.4-7.4-28.9 3.3 24 173.4 246 256.9 381.6 121.3 6.9-7.8-12.6-28.4-20.7-20.4zM213.6 306.6c0 4 4.3 7.3 5.5 8.5 3 3 6.1 4.4 8.5 4.4 3.8 0 2.6.2 22.3-19.5 19.6 19.3 19.1 19.5 22.3 19.5 5.4 0 18.5-10.4 10.7-18.2L265.6 284l18.2-18.2c6.3-6.8-10.1-21.8-16.2-15.7L249.7 268c-18.6-18.8-18.4-19.5-21.5-19.5-5 0-18 11.7-12.4 17.3L234 284c-18.1 17.9-20.4 19.2-20.4 22.6z"
^

SyntaxError: Unexpected token '.', ".4-10.1 89"... is not valid JSON
    at JSON.parse (<anonymous>)
    at /var/www/lib/create-fomantic-icons/lib/parsers/FontAwesome.js:57:43
    at /var/www/lib/create-fomantic-icons/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3)
```

Maybe it is something I am not doing, but as it is prompting for everything and the usage is simply to run the command (after npm install), not sure what to do next.

**Comment by lubber-de** [comment]
Cannot reproduce, works as expected for me (v1.2.1 (also latest develop branch) on node 14):

![image](https://github.com/fomantic/create-fomantic-icons/assets/18379884/c37f0834-af26-44e3-b611-8ee18b86f4f7)

The downloaded zip file might be corrupt (?) Try to delete the file from your temp folder and redownload via script

**Comment by lubber-de** [comment]
I also tried v1.1.0 (in case you used that for FUI < 2.9.0 or SUI), but still works without issues for me
![image](https://github.com/fomantic/create-fomantic-icons/assets/18379884/ced22328-a794-4cab-9a0b-13857d60eda6)

**Comment by ghost** [comment]
That is why asking if I am missing a step perhaps. There is mention of FUI version - do I need to install/download FUI with it? I clone the repo, did npm install, then npm exec/npx... I tested on Windows as well, in case it is some WSL/Linux related thing. Same result..  I tried different versions, different folder...

**Comment by ghost** [comment]
Feel free to assume I am idiot (as I am). but cloning then perhaps left me on a wrong branch and not the release instance? That is something I can think of

**Comment by lubber-de** [comment]
No, you dont need FUI just for this script (this script creates files for a custom FUI build)

- Try a fresh clone or checkout the master/develop branch (doesnt matter atm) 
- cd into the cloned directory
- try yarn (assuming you have yarn installed)
 `yarn start`
 or (without yarn)
`node bin/run`
 or (as stated in the docs)
`npx create-fomantic-icons` (all three work fine for me...)

**Comment by lubber-de** [comment]
Perhaps the downloaded zip file was corrupted by antivirus service? The error message tells about an invalid json key found inside the zip... that's why

**Comment by ghost** [comment]
Hm. My antivirus has never interfered like that before but I did pause it. Cleared the tmp file. Same result though.

I did let it save whatever the string it was trying to parse and it is indeed corrupted - it starts with the sequence:
```
.4-10.1 89.4-64.4 0-61-77.1-89.6-116.9-44.6-23.5 26.4-17.6 42.1-17.6 157.6 50.7 31 118.3 22 160.4-20.1 24.8-24.8 38.5-58 38.5-93 0-35.2-13.8-68.2-38.8-93.3-24.8-24.8-57.8-38.5-93.3-38.5s-68.8 13.8-93.5 38.5c-.3.3-16 16.5-21.2 23.9l-.5.6c-3.3 4.7-6.3 9.1-20.1 6.1-6.9-1.7-14.3-5.8-14.3-11.8V20c0-5 3.9-10.5 10.5-10.5h241.3c8.3 0 8.3 11.6 8.3 15.1 0 3.9 0 15.1-8.3 15.1H130.3v132.9h.3c104.2-109.8 282.8-36 282.8 108.9 0 178.1-244.8 220.3-310.1 62.8zm63.3-260.8c-.5 4.2 4.6 24.5 14.6 20.6C306 56.6 384 144.5 390.6 144.5c4.8 0 22.8-15.3 14.3-22.8-93.2-89-234.5-57-238.3-38.2zM393 414.7C283 524.6 94 475.5 61 310.5c0-12.2-30.4-7.4-28.9 3.3 24 173.4 246 256.9 381.6 121.3 6.9-7.8-12.6-28.4-20.7-20.4zM213.6 306.6c0 4 4.3 7.3 5.5 8.5 3 3 6.1 4.4 8.5 4.4 3.8 0 2.6.2 22.3-19.5 19.6 19.3 19.1 19.5 22.3 19.5 5.4 0 18.5-10.4 10.7-18.2L265.6 284l18.2-18.2c6.3-6.8-10.1-21.8-16.2-15.7L249.7 268c-18.6-18.8-18.4-19.5-21.5-19.5-5 0-18 11.7-12.4 17.3L234 284c-18.1 17.9-20.4 19.2-20.4 22.6z"
      }
    },
    "free": [
      "brands"
    ]
  },
```

So I took the zip from tmp and extracted it to see if it is corrupted. Nope, extracted icons.json is fine. All I can think of then is that for some reason the unzipper library is not working properly for me? Which again, is strange since I tried both Linux and Windows with the same result...

**Comment by lubber-de** [comment]
Does any other font awesome version work for you? (5.x or older 6.x)
In the meantime i tried node 14,15,16,18,20 using yarn and everytime it worked as expected (also using `yarn start --clean` to explicitely delete the temp zip

The unzipper is part of the package.json dependency and i also tried to reinstall those (`yarn install`)...
i unfortunately cannot get this to trigger the error you get :(

which exact node version are you using? perhaps the json.parse is broken there (?)

**Comment by ghost** [comment]
Things tried:

- downgrading unzipper to the min version in package.json
- using clear
- replacing the corrupted json with a clean one

End result: replacing the file manually did not work in WSL. Got a new error:
```
/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:187
  throw generateError(state, message);
  ^
YAMLException: end of the stream or a document separator is expected (36:8)

 33 |     - face-smile-wink
 34 |     - face-surprise
 35 |     - face-tired
 36 |   label: Emoji
-------------^
 37 | energy:
 38 |   icons:
    at generateError (/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:183:10)
    at throwError (/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:187:9)
    at readDocument (/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:1645:5)
    at loadDocuments (/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:1688:5)
    at load (/var/www/lib/create-fomantic-icons/node_modules/js-yaml/lib/loader.js:1714:19)
    at /var/www/lib/create-fomantic-icons/lib/parsers/FontAwesome.js:139:71
    at /var/www/lib/create-fomantic-icons/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
```
Checked that line separator is LF, so not like it is not a match... 

However! Manually replaced the extracted files in Windows - extracted the downloaded zip on top of the existing files. And then it finally ran and worked... Very strange. it is unzipping incorrectly as I extracted manually the same downloaded zip. However why the same method didn't work in WSL/Linux... 😕

**Comment by ghost** [comment]
Actually, I only replaced the one corrupted file in WSL. But it corrupted others as well. Unzipping manually worked there as well. So issue is "solved" - for some reason unzipping corrupts them on my machine somehow while keeping the file sizes the same.. sneaky. 

I personally have Windows 11; in WSL node 20.3.0, npm 9.7.2; in Windows node 18.16.1, npm 9.5.1
Reference: https://github.com/ZJONSSON/node-unzipper/issues/271

**Comment by ghost** [comment]
@lubber-de Me again, Sorry for bothering and thank you again for any help. 

While it did generate the files (same in both WSL and Windows), the list is a bit questionable for me:

- brand-icons.ttf
- brand-icons.woff2
- icons.ttf
- icons.woff2
- outline-icons.ttf
- outline-icons.woff2
- undefined.ttf
- undefined.woff2

From my understanding it shouldn't have the final 2 files. I can guess they might be the "deprecated" icons? 
```
  Results:
    Solid:      1390
    Outline:    163
    Thin:       0
    Brand:      471
    Duotone:    0
    Deprecated: 7
    ----------------
    TOTAL:      2031
```

The second issue is why are there ttf files when the variable file has this: `src: url("@{fontPath}/@{fontName}.woff2") format("woff2") if(@supportIE, e(",") url("@{fontPath}/@{fontName}.woff") format("woff"));` - so I am assuming there needs to be woff files, instead of ttf. There is no reference to ttf files in icon.variables

**Comment by lubber-de** [comment]
> @lubber-de Me again, Sorry for bothering and thank you again for any help.

all fine, but please create a new dedicated issue for different issues rather than reopening unrelated issues again 😉 

> 
> While it did generate the files (same in both WSL and Windows), the list is a bit questionable for me:
> 
> * brand-icons.ttf
> * brand-icons.woff2
> * icons.ttf
> * icons.woff2
> * outline-icons.ttf
> * outline-icons.woff2
> * undefined.ttf
> * undefined.woff2

Reason for this is, that those are copied straight from the "webfonts" directory inside the font awesome zip file.
And the "undefined" files are basically additional files inside the zip package which are FUI irrelevant and have unknown filenames to the script.

You will now recognize that (for fontawesome 6.x) the woff files (not woff2) are still missing. That's because those are not provided inside the fa6.x package anymore.

That said, i fixed both issues in 1.2.2 🙂

**Comment by DonWagner** [comment]
Hey there.

@TivoSoho I have the exact same problem

Maybe I'm the stupid one or maybe it's my bad English but I don't understand what you did to make this work. You are talking about unzipping a zip file and replacing some files in here, but which zip file are you talking about and what files are you replacing?

Sorry if it's obvious but i'm a bit confused
